manifests/install/remove_packages.pp

```diff
@@ -69,14 +67,22 @@
           'pe-ruby-ldap',
         ]
       }
+    } elsif versioncmp("${::aio_agent_version}", "${::package_version}") < 0 {
+      if $::operatingsystem == 'Solaris' {
+        $packages = [
+          'PUPpuppet-agent',
```

Comment:
I think so, solaris is weird. I'll test it out to be sure.

Also, I still need to add SLES/AIX below to get around their weird, non-versionable package managers.